### ISSUE TITLE
unixd/client - discover servers via dns

### ISSUE THREAD

**Issue by Firstyear** [kanidm/kanidm#2667]
Similar to AD sites and services, we could allow discovery of a domain and it's local servers via dns. This would assist with roaming clients, which tend to be the exception not the norm. But it would also eliminate our reliance on load balancers too.

**Comment by Firstyear** [comment]
The barrier here is web browsers probably can't consume it. I think there were some possible attempts though ....

**Comment by Firstyear** [comment]
cc @phoenixbackups

**Comment by Firstyear** [comment]
Probably SRV to start with. But I don't know if https://datatracker.ietf.org/doc/rfc9460/ is widely supported for browsers/dns servers yet, and I think the DNS rfc prohibits https from being a SRV. More research needed :)

**Comment by yaleman** [comment]
Yeah, it's not really something I'd expect to work for browser-side things - how would that ever work from a UX perspective though?

For SRVs I wouldn't say it's a HTTPS service, I'd suggest `_kanidm._tcp.<domain>` with the port set to 443.

**Comment by Gabgobie** [comment]
Just leaving this here for when this becomes relevant ^^

---

I recently went down the browser SRV rabbit hole.

There are these ancient feature requests, all `WONTFIX`:

- <https://bugzilla.mozilla.org/show_bug.cgi?id=14328>
- <https://issues.chromium.org/issues/40309994>
- <https://bugs.webkit.org/show_bug.cgi?id=6872>

Without re-reading them (please take the following with a shovel of salt), I think Mozilla had some worries about security of trusting DNS responses, Chromium was concerned with performance and shifted responsibility to WebKit and they say there's nothing actionable for them - it's the browser's responsibility, they are a layout engine.

---

As for 9460 HTTPS RRs, there is this blog post: <https://kalfeher.com/https-current-state/>

It links a bunch of sources at the bottom of which these seem the most interesting:

- <https://issues.chromium.org/issues/40257146> Implementation in progress
- <https://bugzilla.mozilla.org/show_bug.cgi?id=1634793> Mozilla meta HTTPSSVC in progress

They seem to at least have partial support, Safari is said to have full support.

---

I don't think browsers necessarily need to be part of the equation though. `unixd` discovery should be possible in whichever way you like best, be it a normal SRV record or a misappropriated([?](https://datatracker.ietf.org/doc/html/rfc9460#section-9.6)) HTTPS RR. Browsers would then either ignore the RR and act like they do now or in case of the HTTPS RR perform a more efficient handshake.
